### PR TITLE
add 502 error

### PR DESCRIPTION
**Pull request by alexromano**
for https://tryfinch.atlassian.net/browse/EN-822 adding documentation for the 502 error to indicate upstream provider issues

### CODE REVIEW

File: docs/Development-Guides/Errors/Error-Types.md
Comment (alexromano) [inline docs/Development-Guides/Errors/Error-Types.md:38]: our current naming schema looks like
name: http status name
finch_name: more specific name relating the error to finch

current proposal here is in line with this. what do we think?
would require updating the api-server PR to match this

Comment (eliseuvideira1) [reply docs/Development-Guides/Errors/Error-Types.md:39]: LGTM. Let's wait on Shoabe to weigh in.
I'll wait for this to be merged to change the PR with the finch_code, in case it changes before approve and merge.

**Review by sshariff01 — APPROVED** [review]

File: docs/Development-Guides/Errors/Error-Types.md
Comment (alexromano) [reply docs/Development-Guides/Errors/Error-Types.md:39]: @eliseuvideira1 let's make the changes on the api-server side and merge, and then i will merge this PR while the api-server PR is deploying

**Pull request merged**
